packages/block-editor/src/hooks/position.js

```diff
@@ -203,6 +209,71 @@ export function useIsPositionDisabled( { name: blockName } = {} ) {
 	return ! hasPositionSupport( blockName ) || isDisabled;
 }
 
+function useVisualizer() {
+	const [ property, setProperty ] = useState( false );
+	const { hideBlockInterface, showBlockInterface } = unlock(
+		useDispatch( blockEditorStore )
+	);
+	useEffect( () => {
+		if ( ! property ) {
+			showBlockInterface();
+		} else {
+			hideBlockInterface();
+		}
+	}, [ property, showBlockInterface, hideBlockInterface ] );
+
+	return [ property, setProperty ];
+}
+
+export function PositionVisualizer( { clientId, attributes, forceShow } ) {
+	const positionType = attributes?.style?.position?.type;
+
+	const [ isActive, setIsActive ] = useState( false );
+	const valueRef = useRef( positionType );
+	const timeoutRef = useRef();
+
+	const clearTimer = () => {
+		if ( timeoutRef.current ) {
+			window.clearTimeout( timeoutRef.current );
+		}
+	};
+
+	useEffect( () => {
+		if (
+			! isShallowEqual( positionType, valueRef.current ) &&
+			! forceShow
+		) {
+			setIsActive( true );
+			valueRef.current = positionType;
+
+			timeoutRef.current = setTimeout( () => {
+				setIsActive( false );
+			}, 400 );
+		}
+
+		return () => {
+			setIsActive( false );
+			clearTimer();
+		};
+	}, [ positionType, forceShow ] );
+
+	if ( ! isActive && ! forceShow ) {
+		return null;
+	}
+
+	return (
+		<BlockPopover
```

Comment:
Just a note: To fix the issue where the popover goes off the edge of the screen, it might be worth experimenting with `Popover` instead and then pass it a rect (or `getBoundingClientRect()` function) so that we can ensure it's only rendering over the visible area of the page.

Reply:
Potential fix for this issue in https://github.com/WordPress/gutenberg/pull/49978